src/main.c

```diff
@@ -150,12 +150,76 @@ bool is_appimage(char *path, gboolean verbose)
     return appimage_get_type(path, verbose) != -1;
 }
 
-void *thread_appimage_register_in_system(void *arguments)
-{
-    struct arg_struct *args = arguments;
-    if(is_appimage(args->path, args->verbose) && !appimage_is_registered_in_system(args->path)) {
-        appimage_register_in_system(args->path, args->verbose);
-        update_desktop_set_dirty();
+
+GKeyFile* load_desktop_entry(const char* desktop_file_path) {
+    GKeyFile* key_file_structure = g_key_file_new();
+    gboolean success = g_key_file_load_from_file(key_file_structure, desktop_file_path,
+                                                 G_KEY_FILE_KEEP_COMMENTS | G_KEY_FILE_KEEP_TRANSLATIONS, NULL);
+
+    if (!success) {
+        // Don't remove the dbrackets or the macro will segfault
+        THREADSAFE_G_PRINT("Failed to load the deployed desktop entry, '%s'\n", "a");
+    }
+
+
+    return key_file_structure;
+}
+
+void setup_firejail_on_desktop_entry(GKeyFile* key_file_structure) {
+    char* firejail_exec;
+    char* oldExecValue = g_key_file_get_value(key_file_structure,
+                                              G_KEY_FILE_DESKTOP_GROUP, G_KEY_FILE_DESKTOP_KEY_EXEC, NULL);
+
+    firejail_exec = g_strdup_printf("firejail --env=DESKTOPINTEGRATION=appimaged --noprofile --appimage %s",
+                                    oldExecValue);
+    g_key_file_set_value(key_file_structure, G_KEY_FILE_DESKTOP_GROUP, G_KEY_FILE_DESKTOP_KEY_EXEC, firejail_exec);
+
+    gchar* firejail_profile_group = "Desktop Action FirejailProfile";
+    gchar* firejail_profile_exec = g_strdup_printf(
+        "firejail --env=DESKTOPINTEGRATION=appimaged --private --appimage %s", oldExecValue);
+    gchar* firejail_tryexec = "firejail";
+    g_key_file_set_value(key_file_structure, firejail_profile_group, G_KEY_FILE_DESKTOP_KEY_NAME,
+                         "Run without sandbox profile");
+    g_key_file_set_value(key_file_structure, firejail_profile_group, G_KEY_FILE_DESKTOP_KEY_EXEC,
+                         firejail_profile_exec);
+    g_key_file_set_value(key_file_structure, firejail_profile_group, G_KEY_FILE_DESKTOP_KEY_TRY_EXEC,
+                         firejail_tryexec);
+    g_key_file_set_value(key_file_structure, G_KEY_FILE_DESKTOP_GROUP, "Actions", "FirejailProfile;");
+}
+
+void save_desktop_entry(GKeyFile* key_file_structure, const char* desktop_file_path) {
+    gboolean success = g_key_file_save_to_file(key_file_structure, desktop_file_path, NULL);
+    if (!success) {
+        // Don't remove the dbrackets or the macro will segfault
```

Comment:
Is `dbrackets` a typo or does it have a special meaning?

Reply:
typo, fixed